server/ai_http.go

```diff
@@ -139,8 +155,85 @@ func (h *lphttp) StartLiveVideoToVideo() http.Handler {
 			}
 		}()
 
-		// TODO subscribe to the subscribeUrl for output monitoring
+		// TODO: Use params temporarily to pass the host from gateway
+		host := req.PublishUrl
```

Comment:
Will the `host` be the same for both runner and gateway?

I think the setup will be like the runner is running inside docker and orchestrator is running directly on the host. Then the publish URL should contain `host.docker.internal`? Or will they both run in the host network?

Reply:
I'm having trouble getting this to work correctly. This is why I initially used the ServiceURI of the orchestrator, otherwise we might need to set a startup flag

Reply:
Referring back to [this comment](https://github.com/livepeer/go-livepeer/pull/3263#discussion_r1853282905) - this  is why we should stay close to the original behavior of this code and keep the paths generated by the O relative. At the G, append the host name from sess.Transcoder. At the worker, just before sending down the request to the runner (or within the runner), append host.docker.internal as @leszko mentioned (or the O address in the case of remote workers; presumably the worker already has that info in order to connect to the O).

Reply:
I have updated this logic again by your suggestion and it tested well. On the orchestrator, it will send only the port number to the ai-runner, which uses the client IP from the request to determine the trickle host. This tested well across docker containers and with local/remote IPs